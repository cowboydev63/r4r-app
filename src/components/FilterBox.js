import React from 'react';
import CheckFilter from './CheckFilter';

class FilterBox extends React.PureComponent {
    onChange = value => {
        // TODO: Set filters selected in store, generate new query string, execute new query
        console.log('Dummy filter onChange function firing', {
            param: this.props.facet.param,
            value,
        })
    }

    render() {
        return(
            <div className={`facet__box ${ this.props.className ? 'facet__box--' + this.props.className : '' }`}>
                <h4 className={`facet__title ${ this.props.className ? 'facet__title--' + this.props.className : '' }`}>{ this.props.facet.title }</h4>
                { 
                    this.props.facet.items.map(({
                        key,
                        label,
                        selected,
                        count
                    }, idx) => (
                        <CheckFilter 
                            key={ idx }
                            value={ key }
                            label={ label }
                            checked={ selected }
                            count={ count }
                            onChange={ this.onChange }
                            className={ 'facet__filter'}
                        />
                    ))
                }
            </div>
        )
    }
}

export default FilterBox;