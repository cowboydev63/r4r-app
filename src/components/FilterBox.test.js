import React from 'react';
import { shallow } from 'enzyme';
import FilterBox from './FilterBox';

describe('FilterBox Component', () => {
  it('renders without crashing', () => {
    shallow(<FilterBox />);
  });

  it('renders correctly', () => {
    const wrapper = shallow(<FilterBox />);
    expect(wrapper).toMatchSnapshot();
  })
})