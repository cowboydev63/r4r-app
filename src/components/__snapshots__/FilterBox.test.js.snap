// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBox Component renders correctly 1`] = `""`;
